cumulus_library/template_sql/codeable_concept_denormalize.sql.jinja

```diff
@@ -1,35 +1,47 @@
 CREATE TABLE {{ target_table }} AS (
     WITH
+    {%- for cc_column in cc_columns %}
     {%- for system in code_systems %}
 
-    system_{{ loop.index0 }} AS (
+    system_{{ cc_column.name }}_{{ loop.index0 }} AS (
         SELECT DISTINCT
-            s.id AS id,
+            s.{{ source_id }} AS id,
             '{{ loop.index0 }}' AS priority,
             u.codeable_concept.code AS code,
             u.codeable_concept.display AS display,
             u.codeable_concept.system AS code_system
         FROM
             {{ source_table }} AS s,
-            UNNEST(s.{{ cc_column }}.coding) AS u (codeable_concept) --noqa: AL05
+        {% if cc_column.is_array %}
+            UNNEST(s.{{ cc_column.name }}) AS cc (cc_row),
+            UNNEST(cc.cc_row.coding) AS u (codeable_concept)
+        {% else %}
+            UNNEST(s.{{ cc_column.name }}.coding) AS u (codeable_concept)
+        {%- endif %}
         WHERE
             u.codeable_concept.system = '{{ system }}'
     ), --noqa: LT07
     {%- endfor %}
+    {% endfor %}
```

Comment:
Why does the opening `{%` sometimes have a hyphen and sometimes not?

Reply:
it's jinja's way of determing what to do about continuing whitespace -[ this SO post ](https://stackoverflow.com/questions/11813422/what-does-this-in-jinja2-template-engine-do)is the most parsable explanation i've seen.

Reply:
Of course :roll_eyes:

Reply:
i don't like it either.